### PR TITLE
pkg/terraform/exec/plugins: vendor in azure terraform provider

### PR DESCRIPTION
**Pull request by serbrech**
Vendoring the latest terraform provider in the plugin package

related PR: #1663

### CODE REVIEW

File: pkg/terraform/exec/plugins/azure.go [hunk 1]
```diff
@@ -3,13 +3,27 @@ package plugins
...
 
 func init() {
-	exec := func() {
+	azurermProvider := func() {
 		plugin.Serve(&plugin.ServeOpts{
 			ProviderFunc: azurerm.Provider,
 		})
 	}
-	KnownPlugins["terraform-provider-azurerm"] = exec
+	localProvider := func() {
```
Comment (abhinavdahiya) [inline pkg/terraform/exec/plugins/azure.go]: i would prefer to keep each provider in separate files.

File: pkg/terraform/exec/plugins/Gopkg.toml [hunk 2]
```diff
@@ -52,3 +52,7 @@ ignored = [
 [[override]]
   name = "github.com/hashicorp/go-azure-helpers"
   version = "0.3.2"
+
```
Comment (abhinavdahiya) [inline pkg/terraform/exec/plugins/Gopkg.toml]: you should define the constraint for `github.com/terraform-providers/terraform-provider-local`

Comment (serbrech) [reply pkg/terraform/exec/plugins/Gopkg.toml]: and same with random then. the output from terraform specifically says "any version" when running though.

**Pull request merged**